protostar/git/git_repository.py

```diff
@@ -19,98 +21,56 @@ class Submodule:
     path: Optional[Path] = None
 
 
-# pylint: disable=subprocess-run-check
-
-# set to true when debugging
-GIT_VERBOSE = False
-SHARED_KWARGS = (
-    {"check": True}
-    if GIT_VERBOSE
-    else {
-        "check": True,
-        "stdout": subprocess.DEVNULL,
-        "stderr": subprocess.DEVNULL,
-    }
-)
-
-DEFAULT_CREDENTIALS = [
-    "-c",
-    'user.name="Protostar"',
-    "-c",
-    'user.email="protostar@protostar.protostar"',
-]
-
-
 class GitRepository:
-    def __init__(self, repo_path: Path):
-        """
-        This class should not be instantiated directly, use `Git.init/clone/from_existing` instead.
-        """
-        self.repo_path = repo_path.resolve()
+    @classmethod
+    def create(cls, repo_path: Path):
+        ensure_user_has_git()
+        repo_path.mkdir(parents=True, exist_ok=True)
+        repo = GitRepository(repo_path)
+        repo.init()
+        return repo
 
     @staticmethod
-    def get_repo_root(repo_path: Path):
-        credentials = [] if has_git_credentials() else DEFAULT_CREDENTIALS
-        process = subprocess.run(
-            ["git", *credentials, "rev-parse", "--show-toplevel"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.DEVNULL,
-            cwd=repo_path,
-            check=False,
-        )
+    def from_existing(repo_path: Path):
+        ensure_user_has_git()
+        path = GitRepository.get_repo_root(repo_path)
+        return GitRepository(path)
 
-        if process.returncode:
+    @staticmethod
+    def get_repo_root(repo_path: Path) -> Path:
```

Comment:
This could be an instance method - it'd work like this:

```
GitRepository(repo_path).get_repo_root()
```

But it's not a big deal if we leave it like this

Reply:
Moved this function to `git_helpers.py`.